tensorflow/compiler/xla/service/mlir_gpu/kernel_lowering.cc

```diff
@@ -436,8 +409,10 @@ Status LowerLHLOToGPU(mlir::ModuleOp module,
     tiling_for_unrolling.append(tile_sizes.begin(), tile_sizes.end());
   }
 
-  // First, lower bodies of LHLO operations that contain HLO ops.
-  pm.addPass(absl::make_unique<NestedHloRegionsConverter>());
+  // Legalize from HLO to LHLO.
+  pm.addPass(::mlir::xla_hlo::createLegalizeToLhloPass());
```

Comment:
It is strange to me that we "LegalizeToLhlo" while this function is called "LowerLHLOToGPU", I'd expect this to be a pre-requisite?

Reply:
Yes, you are right. However, the tests contain ```HLO``` operations inside the region of ```LHLO``` operations when it reaches ```LowerLHLOToGPU```.  We decided to avoid duplicating the code in the ```NestedHloRegionsConverter``` since ```createLegalizeToLhloPass```, which implements buffer assignment, is almost identical to ```NestedHloRegionsConverter```. What do you suggest?